Java/src/org/imos/ddb/Test.java

```diff
@@ -30,111 +30,175 @@
-//    System.out.println("lost mem: " + (startFreeMem - connectFreeMem)/1000000 + "Mb");
-    usedMem = Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory();
-    System.out.println("used mem: " + usedMem/1000000 + "Mb");
-    
-    try {
-
-    	List<FieldTrip> trips = mdb.executeQuery("FieldTrip", null, null);
-    	for (FieldTrip t : trips) printObj(t);
-    	
-    	List<DeploymentData> deps = mdb.executeQuery("DeploymentData", null, null);
-    	for (DeploymentData d : deps) printObj(d);
-    	
-    	List<Sites> capeSites = mdb.executeQuery("Sites", null, null);
-    	for (Sites s : capeSites) printObj(s);
-    	
-    	List<CTDData> casts = mdb.executeQuery("CTDData", null, null);
-    	for (CTDData d : casts) printObj(d);
-    	
-    	List<Instruments> inst = mdb.executeQuery("Instruments", null, null);
-    	for (Instruments d : inst) printObj(d);
-    	
-    	List<Sensors> sens = mdb.executeQuery("Sensors", null, null);
-    	for (Sensors d : sens) printObj(d);
-    	
-    	List<InstrumentSensorConfig> instSens = mdb.executeQuery("InstrumentSensorConfig", null, null);
-    	for (InstrumentSensorConfig d : instSens) printObj(d);
-    }
-    catch (Exception e) {e.printStackTrace();}
-    
-    long stopTime = System.currentTimeMillis();
-    System.out.println("Query performed in " + (stopTime - connectTime)/1000 + " seconds.");
-    
-//    endFreeMem = Runtime.getRuntime().totalMemory();
-//    System.out.println("free mem: " + endFreeMem/1000000 + "Mb");
-//    System.out.println("lost mem: " + (connectFreeMem - endFreeMem)/1000000 + "Mb");
-    usedMem = Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory();
-    System.out.println("used mem: " + usedMem/1000000 + "Mb");
-  }
+public class Test
+{
+
+	static void printObj(Object o) throws Exception
+	{
+
+		ArrayList<DBObject> l = (ArrayList<DBObject>) o;
+		for (DBObject i : l)
+		{
+			if (i.o == null)
+				System.out.println(i.name + " = null");
+			else
+				System.out.println(i.name + " class " + i.o.getClass().getName() + " = " + i.o);
+		}
+		// Field []fields = o.getClass().getDeclaredFields();
+
+		// for (Field f : fields) {
+
+		// if (f.isSynthetic()) continue;
+
+		// System.out.println(f.getName() + ": " + f.get(o));
+		// }
+		System.out.println("------");
+	}
+
+	/**
+	 * @param args
+	 */
+	public static void main(String[] args)
+	{
+
+		long startFreeMem, connectFreeMem, endFreeMem;
+		DDB mdb = null;
+
+		String[] jdbcArgs = new String[4];
+
+		//System.out.println("args " + args.length);
+		
+		if (args.length > 0)
+		{
+			if (args.length < 2)
+			{					
+				System.err.println("example usage : org.imos.ddb.Test net.ucanaccess.jdbc.UcanaccessDriver \"jdbc:ucanaccess://C:/Users/jan079/svn/imos-toolbox/TAS-IMOS_DeploymentDatabase2.0.mdb\"");
+				System.err.println("                org.imos.ddb.Test org.postgresql.Driver \"jdbc:postgresql://localhost/IMOS-DEPLOY\" jan079");
+				
+				System.exit(-1);
```

Comment:
Add possible args to the Test class to send driver/connection url from the command line at execution time